### ISSUE TITLE
4.3RC1 with wcslib 7.4: astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Success'

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#11883]
### Description

I am just testing to create a Debian package for 4.3RC1. I get the following error:

```
platform linux -- Python 3.9.2, pytest-6.0.2, py-1.10.0, pluggy-0.13.0

Running tests with Astropy version 4.3-rc1.
Running tests in .pybuild/cpython3_3.9/build.

Date: 2021-06-23T09:27:54

Platform: Linux-5.10.0-7-amd64-x86_64-with-glibc2.31

Executable: /usr/bin/python3.9

Full Python Version: 
3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.19.5
Scipy: 1.6.0
Matplotlib: 3.3.4
h5py: 2.10.0
Pandas: not available
PyERFA: 1.7.2
Cython: 0.29.21
Scikit-image: not available
asdf: not available

Using Astropy options: remote_data: none.

Matplotlib: 3.3.4
Freetype: 2.10.4
ARCH_ON_CI: undefined
IS_CRON: undefined

rootdir: /build/astropy-4.3~rc1, configfile: setup.cfg
plugins: astropy-header-0.1.2, hypothesis-5.43.3, mpl-0.11, doctestplus-0.9.0, arraydiff-0.3, cov-2.10.1, filter-subpackage-0.1.1, remotedata-0.3.2, openfiles-0.5.0
collected 16821 items / 4 errors / 22 skipped / 16795 selected

==================================== ERRORS ====================================
_ ERROR collecting .pybuild/cpython3_3.9/build/astropy/wcs/wcsapi/tests/test_fitswcs.py _
astropy/wcs/wcsapi/tests/test_fitswcs.py:114: in <module>
    WCS_SIMPLE_CELESTIAL = WCS(Header.fromstring(
astropy/wcs/wcs.py:528: in __init__
    self.fix(translate_units=translate_units)
astropy/wcs/wcs.py:707: in fix
    warnings.warn(
E   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Success'.
_ ERROR collecting .pybuild/cpython3_3.9/build/astropy/wcs/wcsapi/tests/test_fitswcs.py _
astropy/wcs/wcsapi/tests/test_fitswcs.py:114: in <module>
    WCS_SIMPLE_CELESTIAL = WCS(Header.fromstring(
astropy/wcs/wcs.py:528: in __init__
    self.fix(translate_units=translate_units)
astropy/wcs/wcs.py:707: in fix
    warnings.warn(
E   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Success'.
_ ERROR collecting .pybuild/cpython3_3.9/build/astropy/wcs/wcsapi/wrappers/tests/test_sliced_wcs.py _
astropy/wcs/wcsapi/wrappers/tests/test_sliced_wcs.py:53: in <module>
    WCS_SPECTRAL_CUBE = WCS(Header.fromstring(HEADER_SPECTRAL_CUBE, sep='\n'))
astropy/wcs/wcs.py:528: in __init__
    self.fix(translate_units=translate_units)
astropy/wcs/wcs.py:707: in fix
    warnings.warn(
E   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Success'.
_ ERROR collecting .pybuild/cpython3_3.9/build/astropy/wcs/wcsapi/wrappers/tests/test_sliced_wcs.py _
astropy/wcs/wcsapi/wrappers/tests/test_sliced_wcs.py:53: in <module>
    WCS_SPECTRAL_CUBE = WCS(Header.fromstring(HEADER_SPECTRAL_CUBE, sep='\n'))
astropy/wcs/wcs.py:528: in __init__
    self.fix(translate_units=translate_units)
astropy/wcs/wcs.py:707: in fix
    warnings.warn(
E   astropy.wcs.wcs.FITSFixedWarning: 'datfix' made the change 'Success'.
=============================== warnings summary ===============================
```
I do not use the wcslib that comes with Astropy, but the Debian package, which is currently (we are in a freeze) older (7.4) than the one provided with Astropy. Therefore I guess these errors are probably temporary until we upgrade after the release of Debian Bullseye. I would recommend to ignore this; the report here is more for completeness.
Also, the error is simple to avoid by adding

```diff
--- a/setup.cfg
+++ b/setup.cfg
@@ -135,6 +135,7 @@ filterwarnings =
        ignore:The toolz.compatibility module is no longer needed:DeprecationWarning
        ignore:parallel reading does not currently work, so falling back to serial
        ignore::DeprecationWarning
+       ignore:'datfix' made the change:astropy.wcs.wcs.FITSFixedWarning
 doctest_norecursedirs = 
        */setup_package.py
 doctest_subpackage_requires = 
```

**Comment by pllim** [comment]
I vaguely remember that warning being a bug from `wcslib`. @dhomeier and @mcara might remember better.

**Comment by dhomeier** [comment]
Yes, fixed in WCSLIB 7.6
https://github.com/astropy/astropy/blob/eb1e8841f51f6c577e884dc55343a92b8f0298e4/cextern/wcslib/CHANGES#L9-L10
